### DISCUSSION TITLE
Questions on exporting disparity

### DISCUSSION THREAD

**Discussion by Ryudoadema** [nagadomi/nunif#160] [Q&A] [answered]
I am wondering when some settings are actually utilized if exporting disparity, then later bringing the edited depthmap back into iw3 for a final conversion. I'm pretty sure depth strength, convergence and video encoding settings aren't used in creating the depth map, but for example Edge Fix and Flicker Reduction. If I want to use those, would I have them on during export disparity and off during final conversion, or vice versa? Would it create issues or double the effects if left on for both steps? Any other important settings that should be turned off or on during either step to not create issues with this sort of process.

**Comment by diverswan9** [comment]
One user I know turns them off for the regurgitation (2nd processing phase). I would probably test both ways and see...

**Comment by nagadomi** [comment, marked as answer]
Export: 
Foreground Scale (mapper) and Edge Fix (edge dilation) are not applied on export, but are applied on import (yml input,  final conversion) instead.

Export disparity:
Foreground Scale (mapper) and Edge Fix (edge dilation) are applied on export and not on import.

YAML file contains its settings.
```
skip_mapper: false
skip_edge_dilation: false
``` 
If `true`, it is not applied on import. 

Flicker Reduction (ema_normalize) is applied on export in both cases, and not applied on import.

**Reply by Ryudoadema** [reply]
Awesome, thanks. So that is the difference between export and export disparity. They both create a yml file though. IW3 somehow knows to apply FS and EF to the "export" yml file when imported back in, and not to apply them when the "export disparity" yml file is imported in the final conversion?

**Reply by nagadomi** [reply]
>They both create a yml file though. IW3 somehow knows 

If yml is created with `--export` option, the following fields are stored in .yml file.
```
skip_mapper: false
skip_edge_dilation: false
```

If yml is created with `--export-disparity`, 
```
skip_mapper: true
skip_edge_dilation: true
```

If it is already applied at export time, skip=true to avoid applying it at import time.

**Reply by nagadomi** [reply]
For more details of the yml file format, see https://github.com/nagadomi/nunif/issues/97#issuecomment-2027349722
